test/acceptance/nouns.ts

```diff
@@ -0,0 +1,202 @@
+import { expect } from "chai";
+import { type BytesLike, namehash, parseEther } from "ethers";
+import { ethers, getNamedAccounts, getUnnamedAccounts } from "hardhat";
+
+import { type ENS } from "../../types";
+import { type NounsToken, NounsToken__factory } from "../../types/ethers-contracts";
+import { type NounsAuctionHouse } from "../../types/ethers-contracts/NounsAuctionHouse";
+import { NounsAuctionHouse__factory } from "../../types/ethers-contracts/factories/NounsAuctionHouse__factory";
+import { ensLabelHash, resolveAddr, resolveText } from "../../utils";
+import { asAccount, mineBlockAtTimestamp } from "../utils";
+
+export function testNounsIntegration(): void {
+  let nounsDomain: string;
+  let nounsHash: BytesLike;
+  let nounsDomainOwner: string;
+  let ens: ENS;
+  let existingTagNode: BytesLike;
+  let nounsAuction: NounsAuctionHouse;
+  let nounsToken: NounsToken;
+
+  describe("Nouns Integration", function () {
+    beforeEach("Set ENSGuilds contract as an ENS manager for the domain", async function () {
+      const { nounsAuctionContract, nounsTokenContract } = await getNamedAccounts();
+      const { ensGuilds, erc721WildcardResolver, flatFeePolicy, openAuthPolicy, ensRegistry } = this.deployedContracts;
+      const { admin } = this.guildInfo;
+      const { ensDefaultResolver: ensDefaultResolverAddr } = await getNamedAccounts();
+
+      // Set up nouns domain
+      nounsDomain = "nouns.eth";
+      nounsHash = namehash(nounsDomain);
+      ens = await ethers.getContractAt("ENS", ensRegistry);
+      nounsDomainOwner = await ens.owner(namehash(nounsDomain));
+      const originalResolver = await ensRegistry.resolver(nounsHash);
+
+      // Set up a subdomain
+      const existingTag = "existing";
+      const existingTagHash = ensLabelHash(existingTag);
+      existingTagNode = namehash(`${existingTag}.${nounsDomain}`);
+
+      await asAccount(nounsDomainOwner, async (signer) => {
+        await ensRegistry
+          .connect(signer)
+          .setSubnodeRecord(nounsHash, existingTagHash, nounsDomainOwner, ensDefaultResolverAddr, 0);
+      });
+
+      // Connect to Nouns Token contract and Nouns Auction House contract
+      nounsToken = NounsToken__factory.connect(nounsTokenContract, ens.runner);
+      nounsAuction = NounsAuctionHouse__factory.connect(nounsAuctionContract, ens.runner);
+
+      // Set up Guild for nouns.eth
+      // 1. Name owner sets ENSGuilds contract as an ENS operator
+      await asAccount(nounsDomainOwner, async (signer) => {
+        await ensRegistry.connect(signer).setApprovalForAll(ensGuilds.getAddress(), true);
+      });
+
+      // 2. Name owner registers nouns.eth as a new guild
+      // const nounsDomain = "nouns.eth";
+      await asAccount(nounsDomainOwner, async (signer) => {
+        // Register guild
+        await ensGuilds
+          .connect(signer)
+          .registerGuild(nounsDomain, admin, flatFeePolicy.getAddress(), openAuthPolicy.getAddress());
```

Comment:
In these tests we should use the same auth policy we'd recommend when the Guild is actually created